modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceAdapter.java

```diff
@@ -525,8 +525,10 @@ public Object getDelegate() {
       return of(
                 create(sink -> ((Reconnectable) sourceInvokationTarget.get()).reconnect(e,
                                                                                         new ReactiveReconnectionCallback(sink))));
+    } else if (sourceInvokationTarget.get() instanceof org.mule.sdk.api.runtime.connectivity.Reconnectable) {
```

Comment:
code shouldn't be checking for two interfaces. All the code should be written in terms of sdk-api reconnectable. Old interface should always be wrapped